certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,160 @@
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+
+    require guardian != otherAccount;
+    bool otherAccountIsGuardian = currentContract.isGuardian(safeContract, otherAccount);
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;
+    require currentGuardiansCount > 0;
+
+    require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
+    require guardianStorageContract.entries[safeContract].guardians[guardian] != guardian;
+
+    currentContract.revokeGuardianWithThreshold(e, safeContract, prevGuardian, guardian, threshold);
+
+    assert safeContract.isModuleEnabled(currentContract);
+    assert e.msg.sender == safeContract;
+    assert !currentContract.isGuardian(safeContract, guardian);
+    assert guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian;
+    assert otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies that the guardian can always be revoked considering ideal conditions.
+rule guardianCanAlwaysBeRevoked(env e, address guardian, address prevGuardian, uint256 threshold) {
 
     requireSocialRecoveryModuleEnabled();
 
-    safeContract.disableModule@withrevert(e, prevModule, currentContract);
+    require e.msg.value == 0;
+    require guardian != 0;
+    require threshold > 0;
+    require guardianStorageContract.countGuardians(safeContract) > threshold;
+    require currentContract.isGuardian(safeContract, guardian);
+    require guardianStorageContract.entries[safeContract].guardians[guardian] != guardian;
+    require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+    require guardianStorageContract.entries[safeContract].guardians[prevGuardian] == guardian;
+
+    require e.msg.sender == safeContract;
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert !isReverted &&
+        guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian &&
+        !currentContract.isGuardian(safeContract, guardian);
+}
+
+// This integrity rule verifies that the addition of a new guardian always reverts if the guardian is already added.
+rule revokeGuardianRevertIfAddressNotGuardian(env e, address otherAccount, address prevGuardian, uint256 threshold) {
+
+    require !currentContract.isGuardian(safeContract, otherAccount);
+
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, otherAccount, threshold);
     bool isReverted = lastReverted;
```

Comment:
unused variable.

Reply:
Removed 👍🏾

Reply:
I don't think it should be removed, but used.

In our other FV spec, Certora recommended that we store the `lastReverted` value in a bool and use it instead of `lastReverted` to not accidentally clobber that value (which can happen if you access a view method in assertions for example)